docs/interactivity.md

```diff
@@ -0,0 +1,483 @@
+            "create_at": 1605082890776,
+            "update_at": 1605082890776,
+            "edit_at": 0,
+            "delete_at": 0,
+            "is_pinned": false,
+            "user_id": "f88nesmr7ifhicr8wwf94oxiwa",
+            "channel_id": "b77x9mu8xindzxkgem8guadsra",
+            "root_id": "",
+            "parent_id": "",
+            "original_id": "",
+            "message": "Here's the body of the post",
+            "type": "",
+            "props": {
+                "disable_group_highlight": true
+            },
+            "hashtags": "",
+            "pending_post_id": "",
+            "reply_count": 0
+        }
+    },
+    "expand": {
+        "app": "",
+        "acting_user": "",
+        "post": "All"
+    }
+}
+```
+
+</details>
+
+<details><summary>Response from App server</summary>
+
+```json
+{
+    "type": "form",
+    "form": {
+        "title": "Send a survey to user",
+        "header": "Message modal form header",
+        "footer": "Message modal form footer",
+        "fields": [
+            {
+                "name": "userID",
+                "type": "user",
+                "description": "User to send the survey to",
+                "label": "User",
+                "hint": "enter user ID or @user",
+                "position": 1,
+                "modal_label": "User"
+            },
+            {
+                "name": "message",
+                "type": "text",
+                "is_required": true,
+                "value": "Here's the body of the post",
+                "description": "Text to ask the user about",
+                "label": "message",
+                "hint": "Anything you want to say",
+                "modal_label": "Text",
+                "subtype": "textarea",
+                "min_length": 2,
+                "max_length": 1024
+            }
+        ]
+    }
+}
+```
+
+</details>
+
+## Submitted Modal
+
+<details><summary>Request from Browser</summary>
+
+`POST` http://localhost:8065/plugins/com.mattermost.apps/api/v1/call
+```json
+{
+    "url": "http://localhost:8065/plugins/com.mattermost.apps/hello/send",
+    "type": "",
+    "expand": {
+        "app": "",
+        "acting_user": "",
+        "post": "All"
+    },
+    "context": {
+        "app_id": "hello",
+        "location_id": "send",
+        "team_id": "gzt3wbyu4tnqfmiqchyatzkx1r",
+        "channel_id": "b77x9mu8xindzxkgem8guadsra",
+        "post_id": "q389yjyqfpygtedmknkannck8c",
+        "root_id": ""
+    },
+    "values": {
+        "userID": "crwxjq4kk7fk3qztocjzenkppc",
+        "message": "Here's the edited body of the post"
+    }
+}
+```
+
+</details>
+
+<details><summary>Response to Browser</summary>
+
+```json
+{"markdown": "Successfully sent survey"}
+```
+
+</details>
+
+<details><summary>Request to App server</summary>
+
+`POST` http://localhost:8065/plugins/com.mattermost.apps/hello/send
+
+Notice `type == ""`, or is missing. This means it is a form submission.
```

Comment:
@levb What are your thoughts on supporting both `type == ""` and `type == "submit"`. I like being able to see `submit` explicitly in the payload. I don't know what the tradeoffs are. We don't want to complicate things on the App side.

Reply:
Agree, let's make "ok" and "submit" explicit, and default to them on ""